arrows/core/video_input_metadata_filter.cxx

```diff
@@ -178,19 +240,63 @@ ::metadata_map()
     return {};
   }
 
-  auto const& map_ptr = m_d->video_input->metadata_map();
-  if( !map_ptr )
+  if( !m_d->metadata_filter )
   {
-    return nullptr;
+    return m_d->video_input->metadata_map();
   }
 
   auto out = vital::metadata_map::map_metadata_t{};
-  for( auto const& i : map_ptr->metadata() )
+
+  if( m_d->filter_uses_image )
   {
-    out.emplace( i.first, this->transform_frame_metadata( i.second ) );
+    if( !m_d->video_input->seekable() )
+    {
+      return nullptr;
```

Comment:
Documentation for `metadata_map()` says 

> If the video is not seekable it will return an empty map.

Confirm that this and other `nullptr` / empty map returns are intentional?

Reply:
Fixed.